builtin/aws/ecs/platform.go

```diff
@@ -814,145 +1026,238 @@ func createALB(
+			tgs[0].Weight = aws.Int64(100)
+		}
+
+		log.Debug("modifying listener to remove target group", "target-groups", len(tgs))
+
+		s.Update("Deregistering this deployment's target group from ALB listener")
+		_, err = elbsrv.ModifyListenerWithContext(ctx, &elbv2.ModifyListenerInput{
+			ListenerArn: listener.ListenerArn,
+			Port:        listener.Port,
+			Protocol:    listener.Protocol,
+			DefaultActions: []*elbv2.Action{
+				{
+					ForwardConfig: &elbv2.ForwardActionConfig{
+						TargetGroups: tgs,
 					},
-					Comment: aws.String("managed by waypoint"),
+					Type: aws.String("forward"),
 				},
-				HostedZoneId: aws.String(albConfig.ZoneId),
-			}
+			},
+		})
+		if err != nil {
+			return fmt.Errorf("failed to modify listener (ARN %q): %s", *listener.ListenerArn, err)
+		}
+		s.Update("Deregistered this deployment's target group from ALB listener")
+	}
 
-			result, err := r53.ChangeResourceRecordSets(input)
-			if err != nil {
-				return nil, nil, err
-			}
-			L.Debug("record created", "change-id", *result.ChangeInfo.Id)
+	s.Done()
+	return nil
+}
 
-			s.Update("Created new Route53 record: %s (zone-id: %s)",
-				albConfig.FQDN, albConfig.ZoneId)
-		}
+func (p *Platform) resourceTargetGroupCreate(
+	ctx context.Context,
+	sg terminal.StepGroup,
+	sess *session.Session,
+	log hclog.Logger,
+	deploymentId DeploymentId,
+	subnets *Resource_Subnets, // Required because we need to know which VPC we're in, and subnets discover it.
+	state *Resource_TargetGroup,
+) error {
+	if p.config.DisableALB {
+		log.Debug("ALB disabled - skipping target group creation")
+		return nil
 	}
 
-	lbArn = listener.LoadBalancerArn
+	s := sg.Add("Initiating target group creation...")
+	defer s.Abort()
 
-	return lbArn, tgArn, err
-}
+	elbsrv := elbv2.New(sess)
 
-func buildLoggingOptions(
-	lo *Logging,
-	region string,
-	logGroup string,
-	defaultStreamPrefix string,
-) map[string]*string {
+	// Use our common deployment ID as the target group name
+	targetGroupName := string(deploymentId)
 
-	result := map[string]*string{
-		"awslogs-region":        aws.String(region),
-		"awslogs-group":         aws.String(logGroup),
-		"awslogs-stream-prefix": aws.String(defaultStreamPrefix),
-	}
+	// We have to clamp at a length of 32 because the Name field
+	// requires that the name is 32 characters or less.
 
-	if lo != nil {
-		// We receive the error `Log driver awslogs disallows options: awslogs-endpoint`
-		// when setting `awslogs-endpoint`, so that is not included here of the
-		// available options
-		// https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_awslogs.html
-		result["awslogs-datetime-format"] = aws.String(lo.DateTimeFormat)
-		result["awslogs-multiline-pattern"] = aws.String(lo.MultilinePattern)
-		result["mode"] = aws.String(lo.Mode)
-		result["max-buffer-size"] = aws.String(lo.MaxBufferSize)
+	// NOTE(izaak): The random part of ULIDs seems to be near the end, so for long app names, we might not get unique names here.
```

Comment:
Leaving this for now because it's not new behavior, and because the future task to replace internal IDs with sequence IDs should help this considerably.